clients/privacy-center/packages/fides-consent/src/lib/consent-banner.ts

```diff
@@ -0,0 +1,553 @@
+  // {"country":"US","ip":"192.168.0.1:1234","location":"US-NY","region":"NY"}
+  // ```
+  geolocationApiUrl?: string
+
+  // Whether or not the banner should be globally disabled
+  isDisabled?: boolean 
+
+  // Whether user geolocation should be enabled. Requires geolocationApiUrl
+  isGeolocationEnabled?: boolean
+
+  // List of country codes where the banner should be enabled. Requires isGeolocationEnabled = true
+  isEnabledCountries?: string[]
+
+  // Display labels used for the banner text
+  labels?: {
+    bannerDescription?: string
+    primaryButton?: string
+    secondaryButton?: string
+    tertiaryButton?: string
+  }
+
+  // URL for the Privacy Center, used to customize consent preferences. Required.
+  privacyCenterUrl?: string
+};
+
+export type UserGeolocation = {
+  country?: string  // "US"
+  ip?: string // "192.168.0.1:12345"
+  location?: string // "US-NY"
+  region?: string // "NY"
+}
+
+// Adapted from https://gist.github.com/henrik/1688572?permalink_comment_id=4317520#gistcomment-4317520o
+// (NOTE: Surprisingly, there's not really a list of these anywhere easily...?)
+const EU_COUNTRY_CODES = [
+  "AT", "BE", "BG", "HR", "CY", "CZ", "DK", "EE", "FI", "FR", "DE", "EL", "HU",
+  "IE", "IT", "LV", "LT", "LU", "MT", "NL", "PL", "PT", "RO", "SK", "SI", "ES",
+  "SE", "UK",
+
+  // ...plus some aliases that might occur (see https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2#Imperfect_implementations)
+  "GB",
+  "GR", 
+
+  // ...plus EEA countries
+  "CH",
+  "IS",
+  "LI",
+  "NO",
+];
+
+/**
+ * Configuration options used for the consent banner. The default values (below)
+ * will be mutated by the banner() function to override with any user-provided
+ * options at runtime.
+ * 
+ * This is effectively a global variable, but we provide getter/setter functions
+ * to make it seem safer and only export the getBannerOptions() one outside this
+ * module.
+ */
+let globalBannerOptions: ConsentBannerOptions = {
+  debug: false,
+  geolocationApiUrl: "http://localhost:3000/location", // TODO: default?
+  isDisabled: false,
+  isGeolocationEnabled: false,
+  isEnabledCountries: EU_COUNTRY_CODES,
+  labels: {
+    bannerDescription: "This website processes your data respectfully, so we require your consent to use cookies.",
+    primaryButton: "Accept All",
+    secondaryButton: "Reject All",
+    tertiaryButton: "Manage Preferences",
+  },
+  privacyCenterUrl: "http://localhost:3000" // TODO: default?
+};
+
+/**
+ * Get the configured options for the consent banner 
+ */
+export const getBannerOptions = (): ConsentBannerOptions => globalBannerOptions;
+
+/**
+ * Wrapper around 'console.log' that only logs output when the 'debug' banner
+ * option is truthy
+ */
+type ConsoleLogParameters = Parameters<typeof console.log>
+const debugLog = (...args: ConsoleLogParameters): void => {
+  if (getBannerOptions().debug) {
+    // eslint-disable-next-line no-console
+    console.log(...args) // TODO: use console.debug instead?
```

Comment:
console.debug doesn't show up in the browser by default, you need to change the log level to verbose to see it. Probably best to keep like this